pkg/auth/providers/saml/saml_actions.go

```diff
@@ -82,55 +85,18 @@ func (s *Provider) testAndApply(actionName string, action *types.Action, request
 	}
 
 	samlConfig = samlConfigApplyInput.SamlConfig
-	s.Request = request
+	p := SamlProviders[s.Name]
+	p.SamlClient.config = &samlConfig
+	p.Request = request
```

Comment:
Discussed offline, but making a note here so that we do not forget it: this is sharing state across api requests. Wont work in HA scenarios and potentially others (like if a user abandons the flow),